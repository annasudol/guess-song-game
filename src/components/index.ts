export * from './alert/Alert';
export * from './button/Button';
export * from './button/ButtonLogout';
export * from './image/Image';
export * from './game/Game';
export * from './login/Login';
export * from './listSongs/ListSongs';
export * from './navigation/Navigation';
export * from './errorMessage/ErrorMessage';
export * from './playlists/Playlists';
export * from './playListItem/PlaylistItem';
export * from './scoreInfo/ScoreInfo';
export * from './scores/Scores';
export * from './summary/Summary';
export * from './search/Search';
export * from './song/Song';
export * from './timer/Timer';
