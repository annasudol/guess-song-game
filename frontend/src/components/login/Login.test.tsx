import { shallow, mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { Login } from '../../components';
jest.mock('react-redux');

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: (): jest.Mock<any, any> => mockDispatch,
}));

describe('component', (): void => {
  describe('Login', (): void => {
    it('match snapshot', (): void => {
      const login = shallow(<Login />);
      expect(toJson(login)).toMatchSnapshot();
    });
    it('calls dispatch', (): void => {
      mount(<Login />);
      expect(mockDispatch).toHaveBeenCalledTimes(1);
    });
  });
});
