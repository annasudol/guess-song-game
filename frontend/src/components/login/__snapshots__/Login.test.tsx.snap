// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Login match snapshot 1`] = `
<div
  className="bg-indigo-900 h-screen w-screen flex justify-center pt-2"
>
  <div
    className="flex flex-col justify-center"
  >
    <h1
      className="font-albaSuper text-center text-orange-500 text-5xl pb-4"
    >
      guess the 
      <br />
      so
      <span>
        ng
      </span>
       game
    </h1>
    <Image
      alt="vinyl player"
      size="large"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "dj-mixer.svg",
        }
      }
    />
    <Button
      href={
        Object {
          "path": "https://accounts.spotify.com/authorize?client_id=9e408fab4a8c4d6096b013b572a8ab1f&redirect_uri=http://localhost:3000/playlist&scope=user-top-read%20user-read-currently-playing%20user-read-playback-state&response_type=token&show_dialog=true",
        }
      }
      spacing={
        Object {
          "mt": "10",
        }
      }
    >
      login to spotify
    </Button>
  </div>
</div>
`;
